src/app/mainview/mainview.component.ts

```diff
@@ -1361,6 +1380,29 @@ export class MainviewComponent implements OnInit, OnDestroy {
             });
     }
 
+    newItem(i){
+        this._nssService.changeItem(i);
+    }
+
+    newRegRegion(regr){
+        this._nssService.changeRegRegion(regr);
+    }
+
+    newStatisticGroup(sg){
+        this._nssService.changeStatisticGroup(sg);
+    }
+
+    newClone(x){
+        this._nssService.changeClone(x);
+    }
+
+    public cloneRowClicked(sg, i, regr) {
+        this.newItem(i);
+        this.newRegRegion(regr);
+        this.newStatisticGroup(sg);
+        this.showCloneScenarioModal();
```

Comment:
Could we possibly combine these into one item to send through to the add scenario component, since we combine them to get the newScenForm anyway? Then we'd only need one subscription in the addscenario component. So, if the user chooses to clone the scenario, we'd combine it into one object and send that to the addScenario modal (using the service and a subject/subscription), which would tell it that it should use that clone. That can then trigger the modal to open, getting rid of lines 1227-1229.

Reply:
Do you know of an easy way to combine these without overriding things like ID, name, code ,ect?

Reply:
I'm not sure what you mean by that, I'd have to look into the code better. What I would start with is looking at where you're combining the scenario in your clonescenario function in the addscenario component and trying to replicate that.

Or, you could just create one object and sending each of these items through as a property, e.g. {newItem: i, newRegRegion: regr, newStatisticGroup: sg}. This is probably not the best "Angular" way to do it, but just showing you a way to combine it and remove all these calls/subscriptions.